server/src/main/java/org/opensearch/env/EnvironmentSettingsResponse.java

```diff
@@ -28,59 +22,41 @@
  * @opensearch.internal
  */
 public class EnvironmentSettingsResponse extends TransportResponse {
-    private Map<Setting<?>, Object> componentSettingValues;
+    private final Settings environmentSettings;
 
-    public EnvironmentSettingsResponse(Settings environmentSettings, List<Setting<?>> componentSettings) {
-        Map<Setting<?>, Object> componentSettingValues = new HashMap<>();
-        for (Setting<?> componentSetting : componentSettings) {
-
-            // Retrieve component setting value from enviornment settings, or default value if it does not exist
-            Object componentSettingValue = componentSetting.get(environmentSettings);
-            componentSettingValues.put(componentSetting, componentSettingValue);
-        }
-        this.componentSettingValues = componentSettingValues;
+    public EnvironmentSettingsResponse(Settings environmentSettings) {
+        this.environmentSettings = environmentSettings;
     }
 
     public EnvironmentSettingsResponse(StreamInput in) throws IOException {
         super(in);
```

Comment:
Nit: I don't like seeing `super(in)` (which in the Response case is a no-op) without a corresponding `out.writeTo()`.  Makes me think too much. :)